src/DataPipes/FillRouteParameterPropertiesDataPipe.php

```diff
@@ -41,7 +43,12 @@ public function handle(
                 continue;
             }
 
-            $properties[$dataProperty->name] = $this->resolveValue($dataProperty, $attribute, $parameter);
+            $properties[$name] = $this->resolveValue($dataProperty, $attribute, $parameter);
+
+            // keep the original property name
+            if ($name !== $dataProperty->name) {
+                $properties[$dataProperty->name] = $properties[$name];
+            }
```

Comment:
Should probably just be

```suggestion
$properties[$dataProperty->outputMappedName ?? $dataProperty->name] = $this->resolveValue($dataProperty, $attribute, $parameter);
```

Instead of the next line

Reply:
Yeah that would seem the best solution, the problem is that validation now chooses the mapped name over the original name (this is a bug I need to fix but it is more complicated then I initially thought) while the data object is created from the original name ...

Something I need time to look at but is more complicated than you would expect since we need to know if a name was mapped for outputting errors.